nilearn/decomposition/multi_pca.py

```diff
@@ -76,15 +83,19 @@ def session_pca(imgs, mask_img, parameters,
             confounds=confounds,
             copy=copy)
     if n_components <= data.shape[0] // 4:
-        U, S, _ = randomized_svd(data.T, n_components)
+        U, S, _ = cache(randomized_svd, memory, memory_level=memory_level,
```

Comment:
Am I getting this right?

randomized_svd does not necessarily return the same result every time it is run?

We cache randomized_svd() now, so that same data returns the same result automatically, without invoking or not random_state aspects within randomized_svd()?

Why is random_state argument not used here, yet it is in a random_svd() call below?